File: settlement/room/service/module/RoomServiceInstance.java
package settlement.room.service.module;

import java.io.Serializable;

import settlement.misc.util.FSERVICE;

public class RoomServiceInstance implements Serializable{

	private static final long serialVersionUID = 1L;
	private short available;
	private short reserved = 0;
	private final short total;
	public byte currentHigh;
	public byte lastHigh;
	
	
	public RoomServiceInstance(int total, RoomService data) {
		this.total = (short) total;
		data.increServices(this.total, 0);
	}
	
	public void report(FSERVICE s, RoomService data, int delta) {
		report(s, data, delta, true);
	}
	
	public void report(FSERVICE s, RoomService data, int delta, boolean load) {
		
		
		
		if(s.findableReservedCanBe()) {
			available += delta;
			data.increServices(0, delta);
			if (delta < 0) {
				data.finder.report(s, -1);
			}else {
				data.finder.report(s, 1);
			}
		}else if (s.findableReservedIs())
			reserved += delta;
		
		if (load) {
			byte h = (byte) (127*(total()-available())/(double)total());
			if (h > currentHigh)
				currentHigh = h;
			if (h > lastHigh)
				lastHigh = h;
		}
	}
	
	
	public void report(FSERVICE s, RoomService data, int delta, int reservable, int reserved) {
			
		this.reserved += reserved*delta;
		available += reservable*delta;
		data.increServices(0, reservable*delta);
		
		if(s.findableReservedCanBe()) {
			if (delta < 0) {
				data.finder.report(s, -1);
			}else {
				data.finder.report(s, 1);
			}
		}
		
		
		
		byte h = (byte) (127*(total()-available())/(double)total());
		if (h > currentHigh)
			currentHigh = h;
		if (h > lastHigh)
			lastHigh = h;
	}
	
	public int available() {
		return available;
	}
	
	public int total() {
		return total;
	}
	
	public int reserved() {
		return reserved;
	}
	
	public double load() {
		return lastHigh/127.0;
	}
	
	public void updateDay() {
		lastHigh = currentHigh;
		currentHigh = 0;
	}
	
	public void clearLoad() {
		lastHigh = 0;
		currentHigh = 0;
	}

	public void dispose(RoomService data) {
		data.increServices(-this.total, -available);
		available = 0;
	}
	
}

File: LegacySource/settlement/room/service/module/RoomServiceInstance.cs
using System;

namespace Settlement.Room.Service.Module
{
    [Serializable]
    public class RoomServiceInstance
    {
        private static readonly long SerialVersionUID = 1L;
        private short available;
        private short reserved = 0;
        private readonly short total;
        public byte CurrentHigh;
        public byte LastHigh;

        public RoomServiceInstance(int total, RoomService data)
        {
            this.total = (short)total;
            data.IncreServices(this.total, 0);
        }

        public void Report(FSERVICE s, RoomService data, int delta)
        {
            Report(s, data, delta, true);
        }

        public void Report(FSERVICE s, RoomService data, int delta, bool load)
        {
            if (s.FindableReservedCanBe())
            {
                available += (short)delta;
                data.IncreServices(0, delta);
                if (delta < 0)
                {
                    data.Finder.Report(s, -1);
                }
                else
                {
                    data.Finder.Report(s, 1);
                }
            }
            else if (s.FindableReservedIs())
            {
                reserved += (short)delta;
            }

            if (load)
            {
                byte h = (byte)(127 * (Total() - Available()) / (double)Total());
                if (h > CurrentHigh)
                {
                    CurrentHigh = h;
                }
                if (h > LastHigh)
                {
                    LastHigh = h;
                }
            }
        }

        public void Report(FSERVICE s, RoomService data, int delta, int reservable, int reserved)
        {
            this.reserved += (short)(reserved * delta);
            available += (short)(reservable * delta);
            data.IncreServices(0, reservable * delta);

            if (s.FindableReservedCanBe())
            {
                if (delta < 0)
                {
                    data.Finder.Report(s, -1);
                }
                else
                {
                    data.Finder.Report(s, 1);
                }
            }

            byte h = (byte)(127 * (Total() - Available()) / (double)Total());
            if (h > CurrentHigh)
            {
                CurrentHigh = h;
            }
            if (h > LastHigh)
            {
                LastHigh = h;
            }
        }

        public int Available()
        {
            return available;
        }

        public int Total()
        {
            return total;
        }

        public int Reserved()
        {
            return reserved;
        }

        public double Load()
        {
            return LastHigh / 127.0;
        }

        public void UpdateDay()
        {
            LastHigh = CurrentHigh;
            CurrentHigh = 0;
        }

        public void ClearLoad()
        {
            LastHigh = 0;
            CurrentHigh = 0;
        }

        public void Dispose(RoomService data)
        {
            data.IncreServices(-this.total, -available);
            available = 0;
        }
    }
}